OutOfSchool/OutOfSchool.WebApi/Startup.cs

```diff
@@ -661,5 +665,70 @@ public static void AddApplicationServices(this WebApplicationBuilder builder)
                 policy.RequireClaim(OpenIddictConstants.Claims.Scope, Constants.OpenIddictScopes.ExternalExportRead);
             });
         });
+
+        // MinIO - Redis notifications
+        builder.Services.AddSingleton<RedisStorageNotificationHandler>();
+        builder.Services.AddSingleton<MinioRedisNotificationBridge>();
+        builder.Services.AddSingleton<IProcessNotificationService, ProcessNotificationService>();
+
+        // Register redis
+        builder.Services.AddSingleton<IConnectionMultiplexer>(provider =>
+        {
+            ConfigurationOptions config = ConfigurationOptions.Parse(redisConnection);
+            config.AbortOnConnectFail = false;
+
+            if (isAPMEnabled)
+            {                
+                var connection = ConnectionMultiplexer.Connect(config);
+                connection.UseElasticApm();
+                return connection;
+            }
+
+            return ConnectionMultiplexer.Connect(config);
+        });
+
+        // Register minio
+        builder.Services.Configure<StorageOptions>(
```

Comment:
Minio is correctly handled in implementation project extensions where all theseinjections should be moved.